app/models/shipit/unknown_repository.rb

```diff
@@ -0,0 +1,23 @@
+module Shipit
+  class UnknownRepository
+    def owner
+      "unknown-owner"
+    end
+
+    def name
+      "unknown-name"
+    end
+
+    def stacks
+      Stack.none
+    end
+
+    def http_url
+      Shipit.github.url("#{owner}/#{name}")
+    end
+
+    def git_url
+      "https://#{Shipit.github.domain}/#{owner}/#{name}.git"
```

Comment:
I'm not sure a NullObject is warranted here. I really like that pattern, but in this case the null values might lead to weird errors, such as trying to clone https://github.com/unknown-owner/unknown-name (which by the way could totally be registered by a malicious user and lead to arbitrary code being executed on your Shipit install).

I'd rather have `Handler#repository` return `nil` or raise an error for now.

If some pattern emerge, and a NullObject would streamline the implementation, then it will always be time to reconsider.

Reply:
How about something a little less clever than the null object? https://github.com/Shopify/shipit-engine/pull/968/commits/fda7be6c60b2439f84df3cdde8b03bc378ee7d50